app/assets/javascripts/hw_combobox/models/combobox/multiple_selection.js

```diff
@@ -0,0 +1,93 @@
+import Combobox from "hw_combobox/models/combobox/base"
+
+Combobox.MultipleSelection = Base => class extends Base {
+  removeSelection(event) {
+    const element = event.target
+    const value = element.getAttribute("data-value")
+    this._commitMultipleSelection({ value }, { selected: false })
+    element.parentElement.remove()
+    this._markInvalid()
+  }
+
+  _connectMultipleSelection() {
+    this._renderSelections()
+    this._setMultipleValue()
+  }
+
+  _renderSelections() {
+    this._multipleSelectionValues().forEach((selectedValue) => {
+      this._renderSelection(selectedValue, this.multipleSelectionsValue[selectedValue])
+    })
+  }
+
+  _renderSelection(value, display) {
+    const selectionWrapper = document.createElement("div")
+    selectionWrapper.id = value
+    selectionWrapper.classList.add("hw-combobox__multiple_selection")
+    const text = document.createElement("div")
+    text.textContent = display
+    const deselector = document.createElement("div")
+    deselector.classList.add("hw-combobox__multiple_selection__remove")
+    deselector.setAttribute("data-action", "click->hw-combobox#removeSelection")
+    deselector.setAttribute("data-value", value)
+    selectionWrapper.appendChild(text)
+    selectionWrapper.appendChild(deselector)
+    this.innerWrapperTarget.insertBefore(selectionWrapper, this.comboboxTarget)
+  }
+
+  _setMultipleValue() {
+    this._setValue(JSON.stringify(this._multipleSelectionValues()))
+  }
+
+  _multipleSelectionValues() {
+    if (this.hasMultipleSelectionsValue) {
+      return Object.keys(this.multipleSelectionsValue)
+    }
+    return []
+  }
+
+  _addSelection(option) {
+    this._commitMultipleSelection(option, { selected: true })
+    this._markValid()
+  }
+
+  _commitMultipleSelection(option, { selected }) {
```

Comment:
I know this function in particular now stands out as messy and linked to the old code; I'm leaving cleanup for when I can return with a clearer mind.

Reply:
Of course - part of it. No worries!